app/src/org/commcare/utils/DateRangeUtils.java

```diff
@@ -0,0 +1,75 @@
+package org.commcare.utils;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+
+import javax.annotation.Nullable;
+
+/**
+ * Utility functions for DateRangePicker widget
+ */
+public class DateRangeUtils {
+
+    // Changing this will require changing this format on ES end as well
+    public static final String DATE_RANGE_ANSWER_PREFIX = "__range__";
+    public static final String DATE_RANGE_ANSWER_DELIMITER = "__";
+    public static final String DATE_RANGE_ANSWER_HUMAN_READABLE_DELIMITER = " to ";
+    private static final String DATE_FORMAT = "yyyy-MM-dd";
+
+    /**
+     * @param humanReadableDateRange human readable fomat for date range as 'startDate to endDate'
+     * @return a Pair of start time and end time that can be supplied to MaterialDatePicker to set a date range,
+     * @throws ParseException if the given humanReadableDateRange is not in 'yyyy-mm-dd to yyyy-mm-dd' format
+     */
+    @Nullable
+    public static androidx.core.util.Pair<Long, Long> parseHumanReadableDate(String humanReadableDateRange) throws ParseException {
+        if (humanReadableDateRange.contains(DATE_RANGE_ANSWER_HUMAN_READABLE_DELIMITER)) {
+            String[] humanReadableDateRangeSplit = humanReadableDateRange.split(DATE_RANGE_ANSWER_HUMAN_READABLE_DELIMITER);
+            if (humanReadableDateRangeSplit.length == 2) {
+                SimpleDateFormat sdf = new SimpleDateFormat(DATE_FORMAT, Locale.US);
+                Date startDate = sdf.parse(humanReadableDateRangeSplit[0]);
+                Date endDate = sdf.parse(humanReadableDateRangeSplit[1]);
+                return new androidx.core.util.Pair<>(getTimeFromDateOffsettingTz(startDate), getTimeFromDateOffsettingTz(endDate));
+            }
+        }
+        throw new ParseException("Argument " + humanReadableDateRange + " should be formatted as 'yyyy-mm-dd to yyyy-mm-dd'", 0);
+    }
+
+    private static Long getTimeFromDateOffsettingTz(Date date) {
```

Comment:
I'm not sure what it returns? Can you see whether a similar utility exists in DataUtils?

Reply:
I wasn't able to find it anywhere else. It removes the timezone offset from the date in millis and return the date. This is because when we set the date to the daterange widget we are setting time in milliseconds without any indication of timezone and the date widget itself doesn't offset the timezone from it to arrive at the offsetted date.